File: Quasics2021Code/MaeTester2022/src/main/cpp/subsystems/Pneumatics.cpp
// Copyright (c) FIRST and other WPILib contributors.
// Open Source Software; you can modify and/or share it under the terms of
// the WPILib BSD license file in the root directory of this project.

#include "subsystems/Pneumatics.h"

#include "Constants.h"

Pneumatics::Pneumatics() {
  SetName("Pneumatics");
  SetSubsystem("Pneumatics");
  std::cout << "---------------------------------\n"
            << "Configuring PCM with CAN ID " << CANBusIds::Other::PCM << "\n"
            << "---------------------------------\n";
  RetractSolenoid();
}

void Pneumatics::Periodic() {
  m_compressor.SetClosedLoopControl(compressorEnabled);
}

void Pneumatics::ExtendSolenoid() {
  SetIntakeSolenoidState(frc::DoubleSolenoid::Value::kForward);
}

void Pneumatics::RetractSolenoid() {
  SetIntakeSolenoidState(frc::DoubleSolenoid::Value::kReverse);
}

void Pneumatics::StopSolenoid() {
  SetIntakeSolenoidState(frc::DoubleSolenoid::Value::kOff);
}

void Pneumatics::ToggleSolenoid() {
  m_intakeSolenoid.Toggle();
#ifdef USE_TWO_SOLENOIDS_FOR_INTAKE
  m_intakeSolenoid2.Toggle();
#endif  // USE_TWO_SOLENOIDS_FOR_INTAKE
}

void Pneumatics::SetIntakeSolenoidState(frc::DoubleSolenoid::Value value) {
  m_intakeSolenoid.Set(value);
#ifdef USE_TWO_SOLENOIDS_FOR_INTAKE
  m_intakeSolenoid2.Set(value);
#endif  // USE_TWO_SOLENOIDS_FOR_INTAKE
}

void Pneumatics::SetCompressorEnabled(bool tf) {
  compressorEnabled = tf;
}

bool Pneumatics::IsCompressorRunning() {
  return m_compressor.Enabled();
}

bool Pneumatics::GetPressureSwitchValue() {
  return m_compressor.GetPressureSwitchValue();
}

double Pneumatics::GetCompressorCurrent() {
  return m_compressor.GetCompressorCurrent();
}

File: Experimental/2022/PneumaticsTesting/src/main/java/frc/robot/subsystems/Pneumatics.java
// Copyright (c) FIRST and other WPILib contributors.
// Open Source Software; you can modify and/or share it under the terms of
// the WPILib BSD license file in the root directory of this project.

package frc.robot.subsystems;

import edu.wpi.first.wpilibj.DoubleSolenoid;
import edu.wpi.first.wpilibj.PneumaticsModuleType;
import edu.wpi.first.wpilibj2.command.SubsystemBase;
import frc.robot.Constants;

public class Pneumatics extends SubsystemBase {
  private final DoubleSolenoid m_doubleSolenoid = new DoubleSolenoid(
      // Hub type: must either be Rev or CTRE
      (Constants.kUsingRevRoboticsPneumaticHub ? PneumaticsModuleType.REVPH : PneumaticsModuleType.CTREPCM),
      // Forward and reverse channels
      Constants.kDoubleSolenoidForwardChannel, Constants.kDoubleSolenoidReverseChannel);

  /** Creates a the subsystem. */
  public Pneumatics() {
  }

  /**
   * Triggers extension of the double solenoid.
   */
  public void extend() {
    m_doubleSolenoid.set(DoubleSolenoid.Value.kForward);
  }

  /**
   * Triggers retraction of the double solenoid.
   */
  public void retract() {
    m_doubleSolenoid.set(DoubleSolenoid.Value.kReverse);
  }
}
